src/transforms/tree.js

```diff
@@ -272,7 +272,7 @@ function parentValue(evaluate) {
 function nameof(path) {
   let i = path.length;
   while (--i > 0) if (slash(path, i)) break;
-  return path.slice(i + 1);
+  return path.slice(i + 1).replaceAll("\\/", "/");
```

Comment:
Yup I just wanted the test to run. But I didn't give this much thought as I'm wondering if we should not use something like \x00 internally for the delimiter, instead of this escaping/unescaping strategy which is bound to give us weird results on unexpected contents.

Reply:
I’d prefer to use “proper” escaping/unescaping than to use \x00, since we don’t want the code to behave strangely if the input happens to have \x00 in it (even though those characters are not displayable).

We need two routines: slashDelimiter to replace the user’s custom delimiter with slashes (while correctly preserving the semantics of slashes and backslashes), and slashUnescape to convert the slash-escaped string to presentation text. These should probably be implemented as for loops/finite state machines rather than using regular expression replacement.